### PR TITLE
save language preference for user

### LINKED ISSUE #1076: Respect user language
As a user, I want to use ephios in my own language. This includes the web interface (language could be determined by browser) and notifications (language needs to be set before)

### PR DESCRIPTION
**Pull request by jeriox**
closes #1076

### CODE REVIEW

File: ephios/core/services/notifications/backends.py [hunk 1]
```diff
@@ -32,6 +33,8 @@ def send_all_notifications():
         for notification in Notification.objects.filter(failed=False):
             if backend.can_send(notification) and backend.user_prefers_sending(notification):
                 try:
+                    if notification.user:
+                        translation.activate(notification.user.language)
                     backend.send(notification)
```
Comment (felixrindt) [inline ephios/core/services/notifications/backends.py:38]: I thought this would be a context manager...

File: ephios/extra/middleware.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+class EphiosLocaleMiddleware(LocaleMiddleware):
+    def process_response(self, request, response):
+        response = super().process_response(request, response)
+        if settings.LANGUAGE_COOKIE_NAME not in request.COOKIES:
+            try:
+                response.set_cookie(settings.LANGUAGE_COOKIE_NAME, request.user.language)
+            except (KeyError, AttributeError):
+                pass
+        return response
```
Comment (felixrindt) [inline ephios/extra/middleware.py:13]: This doesn't set the db field to the browser preference?!

**Review by felixrindt — COMMENTED** [review]
Will need to have a look with my computer too

File: ephios/extra/middleware.py (hunk 2 above)
Comment (jeriox) [reply ephios/extra/middleware.py:13]: no, this stores the saved user preference into the cookie to be used in the session

File: ephios/core/services/notifications/backends.py (hunk 1 above)
Comment (jeriox) [reply ephios/core/services/notifications/backends.py:38]: the django docs don't say that this is possible, they just set the language back to the previous value if needed, but I figured that wouldn't be necessary here

File: ephios/core/forms/users.py [hunk 3]
```diff
@@ -464,3 +464,9 @@ def __init__(self, *args, **kwargs):
 
     def update_preferences(self):
         self.user.preferences["notifications__notifications"] = self.cleaned_data
+
+
+class UserOwnDataForm(ModelForm):
+    class Meta:
+        model = UserProfile
+        fields = ["language"]
```
Comment (felixrindt) [inline ephios/core/forms/users.py:472]: I'd prefer if the label would be "Interface Language" as it does not change the language of user provided values (e.g. event description).

**Review by felixrindt — APPROVED** [review]

File: ephios/core/forms/users.py (hunk 3 above)
Comment (jeriox) [reply ephios/core/forms/users.py]: is now preferred language

**Pull request merged**